src/App.jsx

```diff
@@ -1,58 +1,107 @@
 import 'bulma/css/bulma.css';
 import './App.scss';
 
+import cn from 'classnames';
+import { useState } from 'react';
+
+const SORT_FIELD_ALPHABET = 'name';
+const SORT_FIELD_LENGTH = 'length';
+
 export const goodsFromServer = [
   'Dumplings',
   'Carrot',
   'Eggs',
-  'Ice cream',
   'Apple',
   'Bread',
   'Fish',
   'Honey',
   'Jam',
+  'Ice cream',
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+function getPreparedGoods(goods, { sortField, reverseField }) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((element1, element2) => {
+      switch (sortField) {
+        case SORT_FIELD_LENGTH:
+          return element1[sortField] - element2[sortField];
+
+        case SORT_FIELD_ALPHABET:
+          return element1.localeCompare(element2);
+
+        default:
+          return 0;
+      }
+    });
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  if (reverseField) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverseField, setReverseField] = useState('');
+  const visibleGoods = getPreparedGoods(goodsFromServer,
+    { sortField, reverseField });
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setSortField(SORT_FIELD_ALPHABET)}
+          type="button"
+          className={cn('button is-info',
+            { 'is-light': sortField !== SORT_FIELD_ALPHABET })}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={() => setSortField(SORT_FIELD_LENGTH)}
+          type="button"
+          className={cn('button is-success',
+            { 'is-light': sortField !== SORT_FIELD_LENGTH })}
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={() => setReverseField(!reverseField)}
```

Comment:
change state based on prev state (it should fix failed test). make handler in component and pass reference here
```
const handleReverse = () => setReverseField(prevIsReversed => !prevIsReversed);
```
```suggestion
          onClick={handleReverse}
```
![Screenshot from 2023-08-02 13-07-16](https://github.com/mate-academy/react_list-of-goods-js/assets/33029864/f7510ccc-0313-4cf0-b5d1-722bb072bded)